cmd/query/app/static_handler.go

```diff
@@ -71,12 +73,14 @@ type StaticAssetsHandlerOptions struct {
 	BasePath     string
 	UIConfigPath string
 	LogAccess    bool
+	Features     querysvc.QueryServiceFeature
 	Logger       *zap.Logger
 }
 
 type loadedConfig struct {
-	regexp *regexp.Regexp
-	config []byte
+	regexp                 *regexp.Regexp
+	backendSupportsArchive bool
```

Comment:
this doesn't really belong in this struct. I would instead store the `StorageCapabilities` inside `staticHandler` struct and change the functions below to be on `staticHandler` receiver

Reply:
Indeed, the functions loadAndEnrichIndexHTML and loadUIConfig are dependent on the instance of StaticAssetsHandlerOptions, but I left loadIndexHtml without a receiver as it seems to me completely static.

Done

Reply:
Would you like the refactoring of the receivers to be in a separate commit to facilitate reading, or keep a single commit that contains all the content of the feature?